src/screens/Swap/Connect.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+import React, { useState } from "react";
+import { Trans } from "react-i18next";
+import { StyleSheet } from "react-native";
+import connectManager from "@ledgerhq/live-common/lib/hw/connectManager";
+import { createAction } from "@ledgerhq/live-common/lib/hw/actions/manager";
+import SelectDevice from "../../components/SelectDevice";
+import DeviceActionModal from "../../components/DeviceActionModal";
+import LText from "../../components/LText";
+import colors from "../../colors";
+
+const action = createAction(connectManager);
+
+const Connect = ({ setResult }: { setResult: (result: any) => void }) => {
+  const [device, setDevice] = useState(null);
+  return (
+    <>
+      <LText semiBold style={styles.selectDevice}>
+        <Trans i18nKey={"transfer.swap.selectDevice"} />
+      </LText>
+      <SelectDevice onSelect={setDevice} autoSelectOnAdd />
+      {device ? (
```

Comment:
It's handled by DeviceActionModal internally

Reply:
Fair enough